source/DataHolder/AMDSDataHolder.cpp

```diff
@@ -13,14 +15,32 @@ AMDSDataHolder::~AMDSDataHolder()
 {
 }
 
+AMDSDataHolder& AMDSDataHolder::operator =(AMDSDataHolder &dataHolder)
+{
+	if (this != &dataHolder) {
+		cloneData(&dataHolder);
+	}
+
+	return (*this);
+}
+
+// ======== implementation of AMDSLightWeightDataHolder ===========
 AMDSLightWeightDataHolder::AMDSLightWeightDataHolder(QObject *parent) :
 	AMDSDataHolder(parent)
 {
 	eventData_ = 0;
 }
 
+AMDSLightWeightDataHolder::AMDSLightWeightDataHolder(AMDSLightWeightDataHolder &sourceLightWeightDataHolder, QObject *parent)
+	:AMDSDataHolder(parent)
+{
+	(*this) = sourceLightWeightDataHolder;
+}
+
 AMDSLightWeightDataHolder::~AMDSLightWeightDataHolder()
 {
+	if (eventData_)
```

Comment:
You need to be careful here.  I've been noticing that you've been using `deleteLater()` but you haven't necessarily been setting the pointer to zero.  You'll really want to make sure you do that otherwise the above check would succeed and then you'd get a bad access exception because you're trying to dereference memory that doesn't belong to you anymore.